src/Elasticsearch/Helper/Iterators/SearchHitIterator.php

```diff
@@ -136,4 +141,24 @@ private function readPageData()
         }
 
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function count()
+    {
+        if ($this->count === null) {
+            $this->rewind();
+
+            $this->count = 0;
+
+            $current_page = $this->search_responses->current();
+
+            if (isset($current_page['hits']) && isset($current_page['hits']['total'])) {
+                $this->count = $current_page['hits']['total'];
+            }
```

Comment:
Move these lines to `rewind()` so that you save the count any time you call `rewind()`, and not just when you call `count()`.

``` php
            $this->count = 0;

            if (isset($current_page['hits']) && isset($current_page['hits']['total'])) {
                $this->count = $current_page['hits']['total'];
            }
```